src/policies/exponential_backoff.rs

```diff
@@ -244,8 +249,86 @@ impl ExponentialBackoffBuilder {
             },
         }
     }
-}
 
+    /// Builds an [`ExponentialBackoff`] with the given maximum total duration and calculates max
+    /// retries that should happen applying no jitter.
+    ///
+    /// For example if we set total duration 24 hours, with retry bounds [1s, 24h], we would calculate
+    /// 17 max retries, as 1s * pow(2, 16) = 65536s = ~18 hours and 18th attempt would be way
+    /// after the 24 hours total duration.
+    ///
+    /// If the 17th retry ends up being scheduled after 10 hours due to jitter, [`ExponentialBackoff::should_retry`]
+    /// will return false anyway: no retry will be allowed after total duration.
+    ///
+    /// If one of the 17 allowed retries for some reason (e.g. previous attempts taking a long time) ends up
+    /// being scheduled after total duration, [`ExponentialBackoff::should_retry`] will return false.
+    ///
+    /// Basically we will enforce whatever comes first, max retries or total duration.
+    ///
+    /// Requires the use of [`ExponentialBackoffTimed::for_task_started_at()`].
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use chrono::{DateTime, Utc};
+    /// use retry_policies::{RetryDecision, RetryPolicy};
+    /// use retry_policies::policies::ExponentialBackoff;
+    /// use std::time::Duration;
+    ///
+    /// let exponential_backoff_timed = ExponentialBackoff::builder()
+    ///     .retry_bounds(Duration::from_secs(1), Duration::from_secs(6 * 60 * 60))
+    ///     .build_with_total_retry_duration_and_max_retries(Duration::from_secs(24 * 60 * 60));
+    ///
+    /// assert_eq!(exponential_backoff_timed.max_retries(), Some(17));
+    ///
+    /// let started_at = Utc::now()
+    ///     .checked_sub_signed(chrono::Duration::seconds(25 * 60 * 60))
+    ///     .unwrap();
+    ///
+    /// let should_retry = exponential_backoff_timed.for_task_started_at(started_at).should_retry(0);
+    /// assert!(matches!(RetryDecision::DoNotRetry, should_retry));
+    ///
+    /// let started_at = Utc::now()
+    ///     .checked_sub_signed(chrono::Duration::seconds(1 * 60 * 60))
+    ///     .unwrap();
+    ///
+    /// let should_retry = exponential_backoff_timed.for_task_started_at(started_at).should_retry(18);
+    /// assert!(matches!(RetryDecision::DoNotRetry, should_retry));
+    ///
+    /// ```
+    pub fn build_with_total_retry_duration_and_max_retries(
+        self,
+        total_duration: Duration,
+    ) -> ExponentialBackoffTimed {
+        let mut max_n_retries = None;
+
+        let delays = (0u32..).map(|n| {
+            min(
+                self.max_retry_interval,
+                self.min_retry_interval * 2_u32.checked_pow(n).unwrap_or(u32::MAX),
+            )
```

Comment:
Nit: we could de-dupe this logic so we don't miss it when allowing us to change the base (currently fixed at 2_u32).